bin/tictactoe.rb

```diff
@@ -1,40 +1,152 @@
-#defining the player class object
-class Player
-  attr_accessor :name, :id, :state
-
-  def initialize(name, id, state)
-    @name = name    #player1 or player2
-    @id =id         # 'X' or 'O'
-    @state = state  # win! or lose!
-  end
+puts "Welcome to TIC TAC TOE"
+$turn = 1
 
-  #to display the winner/loser
-  def to_s
-    "#{name} has #{state}"
-  end
+#Registers users to the game
+puts "Player 1 enter your name"
+$player1 = gets.strip
 
-end
+puts "Player 2 enter your name"
+$player2 = gets.strip
 
-#defining the playing board
-class Board
-  @@size = 9
-  attr_accessor :curr_value
+#initializes the game board values
+board = Array.new(9, " ")
 
-  def initialize(curr_value)
-    @curr_value = curr_value
+#method to declare a winner
+def is_gameover?(board)
+  while $turn < 10
+    play(board)
+    if $turn >= 5
+      if hor_check(board) == true
+        break
+      end
+      if ver_check(board) == true
+        break
+      end
+      if dia_check(board) == true
+        break
+      end
+    end
+    if $turn == 10
+      puts "DRAW!"
```

Comment:
All `puts` and `gets` should be inside the `displayController`. The `tictactoe` class is not responsible for user interaction, and so these do not belong here. This is a principle of OOP known as Single Responsibility, which you can read more about [here](https://codeburst.io/understanding-solid-principles-single-responsibility-b7c7ec0bf80)